### ISSUE TITLE
Add non-framework support

### ISSUE THREAD

**Issue by lichen19853** [Giphy/giphy-ios-sdk-ui-example#35]
Can we please support non-dynamic-framework mode in the podfile?

Basically, my project relies on 
- GiphyUISDK: which requires use_frameworks!
- React Native Firebase and bunch other frameworks: which don't work well with use_frameworks!

Wondering if GiphyUISDK can work without use_frameworks!? To repro the issue, remove use_frameworks! from Podfile and do pod install, you'll see an build error: cannot find PINCache

**Comment by cgmaier** [comment]
hey @lichen19853 ! you're using the SDK with react native? we've been wondering how that'd work. 

i'm not sure what you mean by _Wondering if GiphyUISDK can work without GiphyUISDK?_

I'm pretty sure that GiphyUISDK doesn't require the `use_frameworks` flag - it's just the `PINCache` dependency which needs it.

**Comment by lichen19853** [comment]
Thanks @cgmaier . Apologize for the typo, I meant:
"Wondering if GiphyUISDK can work without use_frameworks!?"

So my project uses CocoaPods.
- GiphyUISDK requires use_frameworks flag since its dependency PINCache requires it
- A bunch other pods (mostly React Native pods) doesn't work with use_frameworks flag

As a result, in my project's Podfile, I can't include GiphyUISDK together with other dependencies.

**Comment by cgmaier** [comment]
gotcha! that makes more sense. this seems tricky but hope to have some time to look into it soon 

in the meantime, if you can you install PINCache via Cocoapods without the `use_frameworks!` flag and tell me what happens, that'd be a huge help. would be great to verify this is an issue with PINCache  or with the GIPHY SDK 

thanks for making us aware oof this issue

**Comment by lichen19853** [comment]
Thanks for the quick response.

- I tried including PINCache in Podfile without use_frameworks flag. pod install worked, but project failed with the same error:
Module 'PINCache' not found

- I created a fork to help you see the error
https://github.com/lichen19853/giphy-ios-sdk-ui-example

- Many large projects relied on bunch of dependencies, and they couldn't include use_frameworks flag, thus this ticket can help many other projects as well

**Comment by lichen19853** [comment]
@cgmaier Before we can go without `use_frameworks!` flag, is there a way we can integrat GiphyUISDK without podfile?

**Comment by cgmaier** [comment]
hey @lichen19853 sorry about such a long delay on this. I just removed tried removing the `use_frameworks!` flag, and was able to successfully build. can you give it a try?

**Comment by cgmaier** [comment]
hey @lichen19853 I just tested this again with v1.1.8 and was able to successfully build without `use_frameworks!` 

i'm going to go ahead and close this, but please reopen or create a new issue if you find omitting that flag leads to issues

**Comment by jide** [comment]
I encountered the same issue, and was unable to build without `use_frameworks!`. I tried to add `pod 'PINCache'` to the podfile but got the same error.

Could you provide more details about how you did build without use_frameworks ?

**Comment by filipef101** [comment]
Also can't achieve without use frameworks

**Comment by filipef101** [comment]
:D
add to podfile    pod "PINCache", :modular_headers => true

**Comment by jide** [comment]
:D I ended up getting this right but another bug while compiling misleaded me. Confirming this works.

**Comment by gilbertl** [comment]
@jide Is this still working for you all?

**Comment by gilbertl** [comment]
Thanks @cgmaier . Unfortunately, I added the podspec you provided and ended with the exact error:

"Cannot load underlying module for 'DeepDiff'"

Here's my commit:

https://github.com/gilbertl/giphy-ios-sdk-ui-example/commit/ab338bf9964fcb41e66ccf725f0b4ac392dd6314

**Comment by cgmaier** [comment]
hey ! so odd that it seemed to work for me on fri, though it's possible i was using Xcode 11.1 accidentally. i'll dig in tomorrow

**Comment by gilbertl** [comment]
> hey ! so odd that it seemed to work for me on fri, though it's possible i was using Xcode 11.1 accidentally. i'll dig in tomorrow

@cgmaier , were you able to reproduce the same error?

**Comment by nalexn** [comment]
I'm unable to integrate Giphy for the same reason: `use_frameworks!` is incompatible with the current Pods integration, and if I omit it or replace with `use_modular_headers!` I get these two errors:

```swift
"Cannot load underlying module for 'DeepDiff'"
Failed to load module 'GiphyUISDK'
```
---

So far, integrating Giphy iOS SDK was a very frustrating experience. I already described the problem with Cocoapods, but Carthage integration does not work either because Giphy is built with an old Swift version 5.1

**Comment by jide** [comment]
I could not compile the 2.0 version with this trick, try using 1.3 instead and do https://github.com/Giphy/giphy-ios-sdk-ui-example/issues/35#issuecomment-590925993

**Comment by gilbertl** [comment]
> I could not compile the 2.0 version with this trick, try using 1.3 instead and do [#35 (comment)](https://github.com/Giphy/giphy-ios-sdk-ui-example/issues/35#issuecomment-590925993)

What XCode version are you on?

I switched to 1.3 and continue to get the error

`"Cannot load underlying module for 'DeepDiff'"
Failed to load module 'GiphyUISDK'`

**Comment by jide** [comment]
You shouldn't have a warning for Deepdiff with 1.3, did you try to remove the module completely, clean pod cache, clean project etc ?

**Comment by gilbertl** [comment]
> You shouldn't have a warning for Deepdiff with 1.3, did you try to remove the module completely, clean pod cache, clean project etc ?

Yes. I've 1/ pod cache clean 2/ pod deintegrate  3/ removed the entire repo and start from scratch

Here's the repo: https://github.com/gilbertl/giphy-ios-sdk-ui-example.git . It's pointing to Giphy 1.3.0.

I still get 

`"Cannot load underlying module for 'DeepDiff'" Failed to load module 'GiphyUISDK'`

XCode 11.7


===============================

@cgmaier It's been a week. Any updates on this?

**Comment by jide** [comment]
My bad, I was using version 1.1.9.

**Comment by gilbertl** [comment]
> My bad, I was using version 1.1.9.

I just tried checking out v1.1.9 from the Example repo, commenting out "use_frameworks",   and building from there. It doesn't work. Same issue. Cannot load underlying module for 'DeepDiff.'

**Comment by jide** [comment]
with `pod "PINCache", :modular_headers => true` ?

**Comment by gilbertl** [comment]
> with `pod "PINCache", :modular_headers => true` ?

Yes. With that added.

https://github.com/gilbertl/giphy-ios-sdk-ui-example/tree/gilbertl/non-framework-1.1.9

I'm using Swift. Is that a difference?

**Comment by filipef101** [comment]
I get the same, only after xcode 13.1, with 13.1 works fine somehow

**Comment by gilbertl** [comment]
@cgmaier any resolutions here? Our team is considering implementing our own UI using the SEARCH APIs to unblock ourselves (even if it means losing text and emojis). It's not really an ideal scenario for us or Giphy.

**Comment by filipef101** [comment]
@gilbertl Have you tried Xcode 13.1?

**Comment by filipef101** [comment]
Also don't think this library is official

**Comment by jide** [comment]
I tried your repo and had errors too.

What I can say is that I use react-native-giphy-keyboard with this podspec and it works : 
https://github.com/Gustash/react-native-giphy-keyboard/blob/v1.0.1/react-native-giphy-keyboard.podspec

**Comment by gilbertl** [comment]
> I tried your repo and had errors too.
> 
> What I can say is that I use react-native-giphy-keyboard with this podspec and it works :
> [https://github.com/Gustash/react-native-giphy-keyboard/blob/v1.0.1/react-native-giphy-keyboard.podspec](https://github.com/Gustash/react-native-giphy-keyboard/blob/v1.0.1/react-native-giphy-keyboard.podspec?rgh-link-date=2020-09-11T16%3A11%3A10Z)

I looked at the above repo. The Podfile says

use_frameworks!

https://github.com/Gustash/react-native-giphy-keyboard/blob/87c0e22b78410b76cedd618bb2967033e919d8f7/example/ios/Podfile#L43

**Comment by gilbertl** [comment]
> @gilbertl Have you tried Xcode 13.1?

The latest XCode version is 12
https://en.wikipedia.org/wiki/Xcode

Do you mean iOS 13.1? Are you saying that it'll work if I set the deployment target to iOS >= 13.1 ?

**Comment by jide** [comment]
You should either have use_frameworks! or pod "PINCache", :modular_headers => true.

But none work in your repo example, you could try some of the other options present in the podspec of react-native-giphy-keyboard (maybe static_framework).

**Comment by cgmaier** [comment]
thanks so much for the help @jide! the `pod "PINCache", :modular_headers => true` solution works for me without the `use_frameworks!`. @gilbertl are you still facing issues here?

**Comment by gilbertl** [comment]
> thanks so much for the help @jide! the `pod "PINCache", :modular_headers => true` solution works for me without `use_frameworks!`. @gilbertl are you still facing issues here?

Yes, I'm still seeing issue. Can you share a branch or repo that's working for you?

**Comment by cgmaier** [comment]
I think I understand what's happening now. should have a fix soon

**Comment by cgmaier** [comment]
we are unable to support this request for the foreseeable future.
